util/oidc/oidc.go

```diff
@@ -506,3 +564,33 @@ func createClaimsAuthenticationRequestParameter(requestedClaims map[string]*oidc
 	}
 	return oauth2.SetAuthURLParam("claims", string(claimsRequestRAW)), nil
 }
+
+// GetUserInfoGroups queries the IDP userinfo endpoint for groups claim
+func (a *ClientApp) GetUserInfoGroups(accessToken, issuer, userInfoPath string) ([]string, error) {
+	url := fmt.Sprintf("%s%s", issuer, userInfoPath)
+	request, err := http.NewRequest("GET", url, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	bearer := fmt.Sprintf("Bearer %s", accessToken)
+	request.Header.Set("Authorization", bearer)
+
+	response, err := httpClient.Do(request)
```

Comment:
So, unlike the token response, this response is not signed... which should be fine, as long as the Argo CD <-> Dex connection hasn't been MITM'ed. And if that connection were MITM'ed, then the public keys could be changed anyway, so even the signed token response wouldn't be trustworthy.

Just thinking out the implications. I think these are just reasons to enable HTTPS verification on connections to Dex.  :-P